odxtools/diaglayer.py

```diff
@@ -924,7 +949,20 @@ def decode_response(self, response: Union[bytes, bytearray],
             try:
                 decoded_messages.append(service.decode_message(response))
             except DecodeError as e:
-                pass
+                # check if the response is a global negative response
+                # for the service
+                for gnr in self.global_negative_responses:
+                    try:
+                        decoded_gnr = gnr.decode(response)
+                        decoded_messages.append(
+                            Message(
+                                coded_message=response,
+                                service=service,
+                                structure=gnr,
+                                param_dict=decoded_gnr))
+                    except DecodeError:
```

Comment:
duplicate block?

Reply:
kind of: the hunk above is in the `.decode()` method, while this here for `.decode_response()`. I agree that it's rather inelegant to copy-and paste such a non-trivial piece of code, but generalizing `.decode()` and `.decode_response()` would probably not make things easier to read. (and IMO the de-/encoding logic is quite complicated enough already...)

Reply:
maybe move it to a private function used by both decode / decode response

Reply:
I'll have a look...

Reply:
okay, done: turns out this was way easier than I anticipated and a pretty good idea: 8859c3cbf0c09ea60546de026ee894206f31b7fd